testing/test_hookcaller.py

```diff
@@ -174,7 +174,9 @@ def he_myhook3(arg1):
     assert not pm.hook.he_myhook3.spec.opts["firstresult"]
 
 
-@pytest.mark.parametrize("name", ["hookwrapper", "optionalhook", "tryfirst", "trylast"])
+@pytest.mark.parametrize(
+    "name", ["hookwrapper", "optionalhook", "tryfirst", "trylast", "specname"]
```

Comment:
this ends up making nonsense: `specname=True` / `specname=False` -- I'm not sure this test is doing what you intend?

Reply:
agreed.  created a new test to make sure that functions acquire the appropriate `hookimpl_opt` when using the `specname` option